pkg/kubelet/pod/pod_manager.go

```diff
@@ -153,6 +158,9 @@ func (pm *basicManager) UpdatePod(pod *v1.Pod) {
 // lock.
 func (pm *basicManager) updatePodsInternal(pods ...*v1.Pod) {
 	for _, pod := range pods {
+		if pm.secretManager != nil {
```

Comment:
Let me clarify one thing: "is this function called only when pod changes, or also on some periodic syncs of pods"?
If the former (which I was assuming), then we need to register a pod on every update, as it may have been updated with added/removed secrets.
In that situation, I think my code is correct. The logic of "RegisterPod" is the following:
- take the previous version of a pod
- register all secrets from a current version of a pod (the assuming that pod didn't change secrets, all secrets will have refcount = 2)
- unregister all secrets from a previous version (which will descrease them back to 1)

It's probably suboptimal, but I think it is correct.